src/staking-router-modules/curated-module.service.ts

```diff
@@ -2,94 +2,66 @@ import { Inject, Injectable } from '@nestjs/common';
 
-export interface KeysFilter {
-  used?: boolean;
-  operatorIndex?: number;
-}
-
 @Injectable()
-export class CuratedModuleService {
+export class CuratedModuleService implements StakingModuleInterface {
   constructor(
     @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
     protected readonly keyRegistryService: KeyRegistryService,
     protected readonly keyStorageService: RegistryKeyStorageService,
-    protected readonly metaStorageService: RegistryMetaStorageService,
     protected readonly operatorStorageService: RegistryOperatorStorageService,
-    protected readonly entityManager: EntityManager,
   ) {}
 
+  // we need it
   @Trace(TRACE_TIMEOUT)
-  public async updateKeys(blockHashOrBlockTag: string | number): Promise<void> {
-    await this.keyRegistryService.update(blockHashOrBlockTag);
+  public async update(moduleAddress: string, blockHash: string): Promise<void> {
+    await this.keyRegistryService.update(moduleAddress, blockHash);
   }
 
-  public async getKeyWithMetaByPubkey(pubkey: string): Promise<{ keys: RegistryKey[]; meta: RegistryMeta | null }> {
-    const { keys, meta } = await this.entityManager.transactional(
-      async () => {
-        const keys = await this.keyStorageService.findByPubkey(pubkey.toLocaleLowerCase());
-        const meta = await this.getMetaDataFromStorage();
-
-        return { keys, meta };
-      },
-      { isolationLevel: IsolationLevel.REPEATABLE_READ },
-    );
-
-    return { keys, meta };
+  // we need it
+  public async getCurrentNonce(moduleAddress: string, blockHash: string): Promise<number> {
+    const nonce = await this.keyRegistryService.getNonceFromContract(moduleAddress, blockHash);
+    return nonce;
   }
 
-  public async getKeysWithMetaByPubkeys(
-    pubkeys: string[],
-  ): Promise<{ keys: RegistryKey[]; meta: RegistryMeta | null }> {
-    const { keys, meta } = await this.entityManager.transactional(
-      async () => {
-        const keys = await this.getKeysByPubkeys(pubkeys);
-        const meta = await this.getMetaDataFromStorage();
-
-        return { keys, meta };
-      },
-      { isolationLevel: IsolationLevel.REPEATABLE_READ },
-    );
-
-    return { keys, meta };
-  }
-
-  public async getKeysWithMeta(filters: KeysFilter): Promise<{ keys: RegistryKey[]; meta: RegistryMeta | null }> {
-    const { keys, meta } = await this.entityManager.transactional(
-      async () => {
-        const where = {};
-        if (filters.operatorIndex != undefined) {
-          where['operatorIndex'] = filters.operatorIndex;
-        }
+  public async getKeys(
+    moduleAddress: string,
+    filters: KeysFilter,
+    fields: readonly KeyField[] | undefined,
+  ): Promise<RegistryKey[]> {
+    const where = {};
+    if (filters.operatorIndex != undefined) {
+      where['operatorIndex'] = filters.operatorIndex;
+    }
 
-        if (filters.used != undefined) {
-          where['used'] = filters.used;
-        }
+    if (filters.used != undefined) {
```

Comment:
Why module address is not in filter param.
What for we use where var here. Why not just send do 
`await this.keyStorageService.find(filter, { populate: fields });`